nav2_util/src/lifecycle_node.cpp

```diff
@@ -71,6 +78,29 @@ LifecycleNode::~LifecycleNode()
   }
 }
 
+void LifecycleNode::createBond()
+{
+  RCLCPP_INFO(get_logger(), "Creating bond (%s) to lifecycle manager.", this->get_name());
+
+  bond_ = std::make_unique<bond::Bond>(
+    std::string("bond"),
+    this->get_name(),
+    shared_from_this());
+
+  bond_->setHeartbeatPeriod(0.10);
```

Comment:
@SteveMacenski Can I ask where this value for the period came from? The message is small but on our system running many parts of nav2, this equates to almost 100Hz on the bond topic for 5 nodes.

Reply:
This should be 10hz, not 100hz https://github.com/ros/bond_core/blob/ros2/bondcpp/src/bond.cpp#L291-L299. It was selected to offer a reasonable trade off between responsiveness to a server going down and how long before it goes down would it continue to be unsafe for a robot to operate. Since this needs to match from the lifecycle node -> lifecycle manager and vise versa, this couldn't be easily parameterized so that they would be guaranteed to match, so this is one of the very few "magic numbers" in the stack.

I'd be more than happy to discuss a proposal though to change to another value if you had a suggestion and rationale to share!

Reply:
The stack up of several bond publishers on the same bond topic is giving us the 100 Hz as measured from `ros2 topic hz /bond`. 

Could this value not be kept as a default in a common header file with an associated parameter name?

Reply:
There are not currently any common headers between all of these systems, that I am aware of at least.

You should see 10 * N messages on the topic per second, that is correct, where N is the number of lifecycle bond monitored servers. This should be very attainable for DDS given these messages are very small and the default bringup right now includes composition so its all intraprocess.

Reply:
Is there some precedent for the rate selected from other usage of the bond topic? It seems like there could be a distance metric derived from max speed to come up with a time interval and a timeout value.

Reply:
That would probably break some encapsulation in the servers, since many don't know or care about the robot's speed. What are you trying to solve? 100hz is nothing, that shouldn't be causing any issues even on small platforms!